packages/sandbox/src/file_watching_sandbox.test.ts

```diff
@@ -39,7 +39,8 @@ import {
 
 // Watcher mocks
 const unsubscribeMockFn = mock.fn();
-const subscribeMock = mock.method(watcher, 'subscribe', async () => {
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const subscribeMock = mock.fn(async (_dir, _effect, _options) => {
```

Comment:
```suggestion
const subscribeMock = mock.fn(async () => {
```

that should be fine, if not then something like this: https://github.com/aws-amplify/amplify-backend/blob/1400e6f65cf8e156b8aaeddc4cb37311a37a3fc3/packages/client-config/src/client-config-writer/client_config_writer.test.ts#L27

Reply:
this was to appease the checks like this one where we're checking that subscribe was called with a specific argument, https://github.com/aws-amplify/amplify-backend/pull/2341/files#diff-a3a9ef2ca41c1c9b876d1523afccb0fa5478ffd7ef185395e103467e10bd6484R967

I can pull in the parameters type to give the arguments a type. The underscore satisfies tsc but I also had to add the eslint-disable comment to appease no-unused-vars

Reply:
perhaps
```ts
const subscribeMock = mock.fn(async (..._args: Parameters<typeof subscribe>) => {
  return { unsubscribe: unsubscribeMockFn };
});
```

Reply:
the `mock.fn<>` satisfies both tsc and linter. Let's define parameterless stub AND provides typing for argument assertion.

The `_dir` or `..._args` are not relevant for `mock` . this is for TSC - but that can be handled with `<>`